time.c

```diff
@@ -2871,8 +2871,6 @@ obj2ubits(VALUE obj, size_t bits)
     uint32_t rv;
     int tmp = obj2int(obj);
 
-    if (tmp < 0)
-	rb_raise(rb_eArgError, "argument out of range");
```

Comment:
Is bit-and of a negative value portable?
Probably the initial value of  `u32max` should be `(uint32_t)~0U` too.

Reply:
I _think_ it is portable (even to non-two-complement machines, though I've never used one). We're doing a bit-and on the unsigned cast, and we aren't relying on any specific result of the `&`, just that its value doesn't change (which we know is only possible for the 2**bits values we expect). As long as we don't lose any information in the cast to unsigned (and I think this could have been a problem since int isn't guaranteed to fit in `uint32_t` so we should change it to `unsigned int`) I _believe_ this is fully portable.

> Probably the initial value of u32max should be (uint32_t)~0U too.

Looking at this again I think we can simplify this and set the mask to (1 << bits) - 1